de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java

```diff
@@ -483,8 +499,55 @@ public void onSuccess(ShareResourceLinkDialog result) {
                 }
             });
         } else {
-            diskResourceService.createDataLinks(Arrays.asList(toBeShared.getPath()),
-                                                new CreateDataLinksCallback());
+            diskResourceService.listDataLinks(diskResourceUtil.asStringPathList(Arrays.asList(toBeShared)),
+                                              new DataCallback<String>() {
+
+                                                  @Override
+                                                  public void onFailure(Integer statusCode,
+                                                                        Throwable exception) {
+                                                      ErrorHandler.post(appearance.createDataLinksError(),
+                                                                        exception);
+                                                  }
+
+                                                  @Override
+                                                  public void onSuccess(String result) {
+                                                      List<DataLink> dlList = processResult(result);
+                                                      if (dlList == null || dlList.isEmpty()) {
+                                                          diskResourceService.createDataLinks(Arrays.asList(
+                                                                  toBeShared.getPath()),
+                                                                                              new CreateDataLinksCallback());
+                                                      } else {
+                                                          showPublicLink(dlList);
+                                                      }
+
+                                                  }
+
+                                                  private List<DataLink> processResult(String result) {
```

Comment:
The logic in this method looks almost like a copy of the logic in the `ListDataLinksCallback#onSuccess` method.
Could the common logic in these methods be refactored into a `CallbackConverter` in `DiskResourceServiceFacadeImpl#listDataLinks`? Then the converter could parse a `DataLink` list that these other 2 methods could use directly?